### ISSUE TITLE
Support `iframe` in html block.

### ISSUE THREAD

**Issue by Jocs** [marktext/marktext#925]
### Describe your feature request

Write markdown bellow:

```md
<div>
    <iframe src="http://www.google.com" style="width: 100%; height: 500px;">
</div>
```

and display a iframe in Mark Text.

### What problem does this feature solve? [optional]

Insert iframe in Mark Text.

### Additional context [optional]

Now it only show `Empty HTML Block`

**Comment by fxha** [comment]
This is due DOMPurify and GFM Spec `6.11 Disallowed Raw HTML`:

> GFM enables the tagfilter extension, where the following HTML tags will be filtered when rendering HTML output:
> - <iframe>
> [...]

I don't know how secure it is to enable `iframe` because you can load JavaScript into Mark Text.

**Comment by Jocs** [comment]
According to my judgment, there should be no security problems. After all, the iframe is in a separate javascript execution environment and can only communicate with the outside window via `postMessage`. But the standard does not support `iframes`. I think we can add a user preference option, whether it supports iframe elements, style attributes, etc.

**Comment by sippeangelo** [comment]
I use Mark Text to edit my blog and I frequently insert embedded YouTube videos, which happen to be iframes. It would be nice to see them!

An option to disable DOMPurify on a document by document basis would be nice, similar to how editors like Microsoft Word will warn you about embedded scripts in documents and only letting them run if the user chose to trust them.

**Comment by Jocs** [comment]
>An option to disable DOMPurify on a document by document basis would be nice, similar to how editors like Microsoft Word will warn you about embedded scripts in documents and only letting them run if the user chose to trust them.

Good idea, I think a user preference setting to enable `iframes` in Mark Text is also OK.

How to embedded YouTube videos, can you show me an example of markdown content?

**Comment by sippeangelo** [comment]
> How to embedded YouTube videos, can you show me an example of markdown content?

My site generator doesn't do any filtering so it simply passes any HTML through. This is how my document looks for example:
```markdown
# Trailer

<pre><iframe width="100%" height="477" src="https://www.youtube.com/embed/C0zoI3RJ_fw" frameborder="0" rel="0" allow="encrypted-media" allowfullscreen></iframe></pre>
```

**Comment by Dd-ouni** [comment]
Has this been added

**Comment by Dd-ouni** [comment]
> > How to embedded YouTube videos, can you show me an example of markdown content?如何嵌入 YouTube 视频，你能给我一个减价内容的例子吗？
> 
> My site generator doesn't do any filtering so it simply passes any HTML through. This is how my document looks for example:
> 
> 我的网站生成器不做任何过滤，所以它只是简单地传递任何 HTML。例如，我的文档是这样的:
> 
> ```gfm
> # Trailer
> 
> <pre><iframe width="100%" height="477" src="https://www.youtube.com/embed/C0zoI3RJ_fw" frameborder="0" rel="0" allow="encrypted-media" allowfullscreen></iframe></pre>
> ```

How to solve it

**Comment by fxha** [comment]
iframe's aren't supported in MarkText and also not planned due to security reasons. All insecure HTML elements are hidden during preview but not removed from the markdown document.

**Comment by mangoepic** [comment]
> iframe's aren't supported in MarkText and also not planned due to security reasons. All insecure HTML elements are hidden during preview but not removed from the markdown document.

What about an option to let the user choose if they want to take the risk with a certain block? Like an informing popup with cancel confirm like the windows smart screen thing? I'd love to embed youtube videos or soundcloud, discord also made it possible somehow

**Comment by Jocs** [comment]
Thank you for the request. Rendering &lt;iframe&gt; elements in the editor preview is not supported and is not planned. DOMPurify removes iframes by default because they can load arbitrary external content and scripts, creating a significant XSS and data-exfiltration risk in a local Markdown editor. The GFM spec also lists iframe in its disallowed-raw-HTML tagfilter for the same reason. The current engine deliberately locks this behavior with an explicit test to prevent future regressions. The iframe tags are preserved in your Markdown source file and will pass through unmodified to any downstream renderer that chooses to allow them. We are closing this issue as out-of-scope for MarkText's security model.
